src/kvstore/LogEncoder.cpp

```diff
@@ -164,6 +164,63 @@ std::vector<folly::StringPiece> decodeMultiValues(folly::StringPiece encoded) {
     return values;
 }
 
+std::string encodeBatchValue(const std::vector<std::pair<BatchLogType,
+                             std::pair<std::string, std::string>>>& batch) {
+    auto type = LogType::OP_BATCH_WRITE;
+    std::string data, encoded;
```

Comment:
Why use data here?  just encoded is enough.
Before append data into "encoded",  please reserve enough space for it.